### ISSUE TITLE
OnTraceHit + Async Node: Pass the entire HitResult

### ISSUE THREAD

**Issue by matanshukry** [rlewicki/MeleeTrace#20]
# Description

Currently only the ImpactPoint, ImpactNormal, and BoneName are passed to the OnTraceHit events, and as such to the AsyncNode as well.

However we already have the entire hit result ( [link](https://github.com/rlewicki/MeleeTrace/blob/master/Source/MeleeTrace/Private/MeleeTraceComponent.cpp#L95C6-L95C16) ) , and it has info that could be useful - so would be great if we can pass that instead 🙏

**Comment by rlewicki** [comment]
Yeah I guess we could just do that. Not sure why I decided not to do that back then. Probably had a reason for that but can't remember what that was. Will keep track of that one. Thanks for creating the issue!
